File: git/demo/backend/src/main/java/streamer_website/demo/entity/twitch/TwitchChannel.java
package streamer_website.demo.entity.twitch;

import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;

@Entity
@Table(name = "twitch_channels")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TwitchChannel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "twitch_user_id", nullable = false, unique = true, length = 128)
    private String twitchUserId;

    @Column(name = "user_name", nullable = false, length = 128)
    private String userName;

    @Column(name = "is_active", nullable = false)
    private boolean isActive = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt = Instant.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = Instant.now();
    }
}

File: git/demo/backend-go/internal/domain/twitch_channel.go
package domain

import (
	"time"
)

type TwitchChannel struct {
	ID           int64     `json:"id"`
	TwitchUserID string    `json:"twitch_user_id"`
	Username     string    `json:"username"`
	IsActive     bool      `json:"is_active"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

func NewTwitchChannel(twitchUserID, username string) *TwitchChannel {
	now := time.Now()
	return &TwitchChannel{
		TwitchUserID: twitchUserID,
		Username:     username,
		IsActive:     true,
		CreatedAt:    now,
		UpdatedAt:    now,
	}
}

func (c *TwitchChannel) Update(username string) {
	c.Username = username
	c.UpdatedAt = time.Now()
}

func (c *TwitchChannel) Activate() {
	c.IsActive = true
	c.UpdatedAt = time.Now()
}

func (c *TwitchChannel) Deactivate() {
	c.IsActive = false
	c.UpdatedAt = time.Now()
}